### PR TITLE
MISUV-6581 - TxM event -Create Income Source: success/failure (API#1566)

### PR DESCRIPTION
**Pull request by JordHemsworth**

### CODE REVIEW

File: app/audit/Utilities.scala [hunk 1]
```diff
@@ -35,6 +35,15 @@ object Utilities {
     ("credId", user.credId) ++
     userType(user.userType)
 
+  def userAuditDetailsNino(user: MtdItUserBase[_]): JsObject = Json.obj(
+    "nino" -> user.nino,
+    "mtditid" -> user.mtditid
+  ) ++
+    ("agentReferenceNumber", user.arn) ++
```
Comment (joesheridan) [inline app/audit/Utilities.scala:42]: dupe code here from above - can you extract a common private method eg getBaseDetails (without nino fields) and then apply either nino or nationalinsurancenumber depending on the called method (ie userAuditDetails or userAuditdetailsNino)?

File: app/audit/models/CreateIncomeSourceAuditModel.scala [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+import audit.Utilities.{userAuditDetails, userAuditDetailsNino}
+import auth.MtdItUser
+import enums.IncomeSourceJourney.{IncomeSourceType, SelfEmployment}
+import models.createIncomeSource.CreateIncomeSourceResponse
+import models.incomeSourceDetails.viewmodels.{CheckBusinessDetailsViewModel, CheckDetailsViewModel, CheckPropertyViewModel}
+import play.api.libs.json.{JsObject, JsValue, Json}
+import utils.Utilities._
+
+
```
Comment (joesheridan) [inline app/audit/models/CreateIncomeSourceAuditModel.scala:27]: could fix: two spacing lines is unnecessary - 1 is always enough 😄

File: app/audit/models/CreateIncomeSourceAuditModel.scala [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        seDetails
+
+      case propertyViewModel: CheckPropertyViewModel =>
+        val propDetails = Json.obj("dateStarted" -> propertyViewModel.tradingStartDate)
+
+        propDetails
+    }
+
+    val accountingMethod = if (viewModel.cashOrAccruals.toLowerCase.equals("cash")) {
+      "Cash basis accounting"
```
Comment (joesheridan) [inline app/audit/models/CreateIncomeSourceAuditModel.scala:82]: instead of "cash basis accounting" - can we preserve the API vals eg CASH or ACCRUALS?

Comment (JordHemsworth) [reply app/audit/models/CreateIncomeSourceAuditModel.scala:82]: We can however the example json from ticket does have -   "accountingMethod":"cash basis accounting"   
Are we better keeping the vals?

File: app/audit/Utilities.scala (hunk 1 above)
Comment (JordHemsworth) [reply app/audit/Utilities.scala:32]: resolved in recent commit

File: app/audit/models/CreateIncomeSourceAuditModel.scala (hunk 2 above)
Comment (JordHemsworth) [reply app/audit/models/CreateIncomeSourceAuditModel.scala]: good eye

File: app/audit/models/CreateIncomeSourceAuditModel.scala (hunk 3 above)
Comment (JordHemsworth) [reply app/audit/models/CreateIncomeSourceAuditModel.scala]: resolved in recent commit

**Pull request merged**